harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -708,8 +715,8 @@ def _save(self, path: pathlib.Path) -> workload.Response:
             "optimizer_state_dict": self.context.optimizer.state_dict(),
         }
 
-        if self.lr_helper:
-            checkpoint["lr_scheduler"] = self.lr_helper.state_dict()
+        if self.context.lr_scheduler:
```

Comment:
non-blocking: use `if self.context.lr_scheduler is not None`, the way that it's done in other lines of this PR.  Also I just prefer being explicit about a None check.